rdr_service/services/data_dictionary_updater.py

```diff
@@ -0,0 +1,287 @@
+from sqlalchemy import MetaData
+from sqlalchemy.orm.attributes import InstrumentedAttribute
+from sphinx.pycode import ModuleAnalyzer
+from sqlalchemy.orm import joinedload
+
+from rdr_service.model.base import Base
+from rdr_service.model.code import Code
+from rdr_service.model.hpo import HPO
+from rdr_service.model.questionnaire import QuestionnaireHistory
+from rdr_service.model.questionnaire_response import QuestionnaireResponse
+from rdr_service.model.site import Site
+from rdr_service.services.google_sheets_client import GoogleSheetsClient
+
+dictionary_tab_id = 'RDR Data Dictionary'
+internal_tables_tab_id = 'RDR Internal Only'
+hpo_key_tab_id = 'Key_HPO'
+questionnaire_key_tab_id = 'Key_Questionnaire'
+site_key_tab_id = 'Key_Site'
+
+
+class DataDictionaryUpdater:
+    def __init__(self, gcp_service_key_id, dictionary_sheet_id, session):
+        self.gcp_service_key_id = gcp_service_key_id
+        self.dictionary_sheet_id = dictionary_sheet_id
+        self.session = session
+
+        # List out tables that should be marked as internal, but aren't mapped in the ORM
+        self.internal_table_list = [
+            'alembic_version',
+            'metrics_age_cache_bak',
+            'metrics_enrollment_status_cache_bak',
+            'metrics_gender_cache_bak',
+            'metrics_lifecycle_cache_bak',
+            'metrics_race_cache_bak',
+            'metrics_region_cache_bak',
+            'temp_debug_bigquery_sync',
+            'tmp_participants'
```

Comment:
Might want to add 'ptsc_cohort' to this list.

Reply:
Good idea, thanks for noticing it.